watcherx/websocket_client.go

```diff
@@ -0,0 +1,51 @@
+package watcherx
+
+import (
+	"context"
+	"net/url"
+
+	"github.com/gorilla/websocket"
+	"github.com/pkg/errors"
+)
+
+func WatchWebsocket(ctx context.Context, u *url.URL, c EventChannel) error {
+	conn, _, err := websocket.DefaultDialer.Dial(u.String(), nil)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+	go closeOnDone(ctx, conn)
+	go forwardWebsocketEvents(conn, c, u)
+	return nil
+}
+
+func closeOnDone(ctx context.Context, conn *websocket.Conn) {
+	<-ctx.Done()
+	conn.Close()
+}
+
+func forwardWebsocketEvents(conn *websocket.Conn, c EventChannel, u *url.URL) {
+	serverURL := source(u.String())
+	defer conn.Close()
+	for {
```

Comment:
This queue never finishes. Maybe listen on `context.Done()`?

Reply:
That's why there is https://github.com/ory/x/pull/190/files#diff-5db460a2bf191856555ebe1ceb20c2b3R16

See https://github.com/gorilla/websocket/issues/71 as a reference.

Reply:
`conn.ReadMessage` is blocking

Reply:
So `WatchWebsocket` will just accumulate go routines that can not be killed? Not sure if I like that :D